### PR TITLE
Set the bytecode index for the instrumentation in goto_check.cpp

### PR DESCRIPTION
**Pull request by cristina-david**
This fixes~ diffblue/test-gen#243

### CODE REVIEW

File: src/analyses/goto_check.cpp [hunk 1]
```diff
@@ -1820,6 +1820,11 @@ void goto_checkt::goto_check(
 
         if(it->source_location.get_column()!=irep_idt())
           i_it->source_location.set_column(it->source_location.get_column());
+
+        if(mode==ID_java &&
```
Comment (peterschrammel) [inline src/analyses/goto_check.cpp]: Is the mode check needed?

File: src/analyses/goto_check.cpp [hunk 2]
```diff
@@ -1820,6 +1820,11 @@ void goto_checkt::goto_check(
...
+           it->source_location.get_java_bytecode_index()!=irep_idt())
```
Comment (peterschrammel) [inline src/analyses/goto_check.cpp]: ! .empty()?

Comment (cristina-david) [inline src/analyses/goto_check.cpp]: In principle yes, but I wanted to maintain the style used in the previous lines. I could potentially change them all in a different PR.

**Comment by cristina-david** [comment]
@peterschrammel I've addressed one of your comments. For the second one I could do it in a different PR for all the conditionals in goto_check.cpp. Otherwise, it will be inconsistent.

**Pull request merged**